xray/audit/yarn/yarn.go

```diff
@@ -34,12 +34,14 @@ func BuildDependencyTree() (dependencyTree []*xrayUtils.GraphNode, err error) {
 		return
 	}
 	// Calculate Yarn dependencies
-	dependenciesMap, _, err := biUtils.GetYarnDependencies(executablePath, currentDir, packageInfo, log.Logger)
+	dependenciesMap, root, err := biUtils.GetYarnDependencies(executablePath, currentDir, packageInfo, log.Logger)
 	if err != nil {
 		return
 	}
+	// In some versions of Yarn, the version of the root package is '0.0.0-use.local', instead of the version in the package.json file.
+	root.Details.Version = packageInfo.Version
```

Comment:
we already replacing it for the case mentioned in comment, this replacement here always happens and can cause mismatch when building the graph